graphql/resolvers/addComment.ts

```diff
@@ -11,6 +11,135 @@ import {
+import { MessageResponse } from '../../@types/discordBot'
+
+type CommentWithAuthorAndSubmission = Comment & {
+  author: {
+    id: number
+    username: string
+    discordId: string | null
+    name: string
+  }
+  submission: Submission & {
+    user: { id: number; username: string; discordId: string | null }
+    lesson: Lesson
+    challenge: Challenge
+  }
+}
+
+type SendNotificationsParams = {
+  line: InputMaybe<number> | undefined
+  fileName: InputMaybe<string> | undefined
+  submissionId: number
+  comment: CommentWithAuthorAndSubmission
+}
+
+const sendNotifications = async ({
+  line,
+  submissionId,
+  fileName,
+  comment
+}: SendNotificationsParams) => {
+  const restOfComments = await prisma.comment.findMany({
+    where: {
+      line,
+      submissionId,
+      fileName
+    },
+    select: {
+      author: {
+        select: {
+          id: true,
+          discordId: true
+        }
+      }
+    },
+    distinct: ['authorId']
+  })
+
+  const { submission, author } = comment
+
+  const notificationEmbed = (forSubmissionOwner: boolean): APIEmbed => {
+    const params = {
+      title: forSubmissionOwner
+        ? `New comment by a reviewer on submission`
+        : `New comment on submission`,
+      description: forSubmissionOwner
+        ? 'commented on your submission to the'
+        : 'added a comment on'
+    }
+
+    return {
+      color: PRIMARY_COLOR_HEX,
+      title: params.title,
+      url: `${CURRICULUM_URL}/${submission.lesson.slug}`,
+      thumbnail: {
+        url: getLessonCoverPNG(submission.lesson.order)
+      },
+      author: {
+        name: author.name,
+        url: `${PROFILE_URL}/${author.username}`,
+        icon_url: C0D3_ICON_URL
+      },
+      description: `${getDiscordMessageUserIdString(author)} ${
+        params.description
+      } challenge **${submission.challenge.title}**${
+        comment.line ? ` on **Line ${comment.line}**` : ''
+      }.`,
+      fields: [
+        {
+          name: 'Comment',
+          value: comment.content
+        }
+      ]
+    }
+  }
+
+  // sends a notification to the submission author
+  const sendNotificationToAuthor = async () => {
+    if (author.id !== submission.user.id && submission.user.discordId) {
+      await sendDirectMessage(
+        submission.user.discordId,
+        '',
+        notificationEmbed(true)
+      )
+    }
+  }
+
+  // sends a notification to all of those who commented on
+  // the same submission line except the comment author and the student who made the submission
+  const sendNotificationsToParticipants = async () => {
+    const notifications = restOfComments.reduce(
+      (acc: Promise<MessageResponse>[], participant) => {
+        const isCommentAuthor = participant.author.id === author.id
+        const isSubmissionAuthor = participant.author.id === submission.user.id
+        const hasDiscordId = participant.author.discordId
+
+        if ((!isCommentAuthor || !isSubmissionAuthor) && hasDiscordId) {
```

Comment:
This condition is not correct. `if ((user is not comment author or user is not submission author) and they have a discord id) send notification` so if I am not the submission author, but I am the comment author, I'll still get notified which should not happen since I know I made the comment.